lib/application.js

```diff
@@ -124,6 +124,9 @@ app.model = function(Model, config) {
   if (arguments.length > 1) {
     config = config || {};
     configureModel(Model, config, this);
+    this.on('modelRemoted', function() {
+      setSharedMethodSharedProperties(Model, this, config);
```

Comment:
This will add one listener for each application model, which may trigger maxListener warning.

Is there a way how to install a single listener which will walk through application's models to apply "sharedMethods" configuration?

Reply:
Hey, sorry for the delay.

> So the problem was that setSharedMethodSharedProperties wasn't exported anywhere. This made it difficult to call it from somewhere useful and setup only one watcher. I've moved it to its own file so it can be called from loopback.js and application.js. Let me know if you can think of a better solution.

I like that 👍 

> The other issue is that the model config only lives for the duration of app.model run. I stuck it on the Model.config property so I could get to it from loopback.js but this is probably not the best way to do it (and is causing another test to fail). Is there a better way to make the config available on the model itself at a later time?

I am proposing to include the config in the arguments of `modelRemoted` event, see https://github.com/strongloop/loopback/blob/6ba35c297b178aaa24272b7b04fa4eb3d7f26e41/lib/application.js#L149

```js
// my proposal
this.emit('modelRemoted', Model.sharedClass, config);
```

Reply:
>I am proposing to include the config in the arguments of modelRemoted event, see
>`this.emit('modelRemoted', Model.sharedClass, config);`

The problem I see here is that that only emits the config for the model being remoted. We need to re-configure every model in the app and - therefore - get at the config for every model somehow. This is why I was storing it on the models individually. Maybe it should be a sub-property of `sharedClass` ie `Model.sharedClass.config`?

Reply:
You are right, I did not realize that all classes need to be reconfigured. Let me take a closer look to better understand why some tests are failing with your current approach.